src/magicbox/ResultPreviewsGrid.ts

```diff
@@ -0,0 +1,268 @@
+import { $$, Dom } from '../utils/Dom';
+import { defaults, findIndex } from 'underscore';
+import { Component } from '../ui/Base/Component';
+import { ISelectableItemsContainer } from './SuggestionsManager';
+
+export enum ResultPreviewsGridDirection {
+  Up = 'Up',
+  Down = 'Down',
+  Left = 'Left',
+  Right = 'Right'
+}
+
+export interface ISearchResultPreview {
+  element: HTMLElement;
+  onSelect?: () => void;
+}
+
+interface IActiveSearchResultPreview extends ISearchResultPreview {
+  deactivate: () => void;
+}
+
+export interface IResultPreviewsGridOptions {
+  selectedClass?: string;
+}
+
+export interface IPreviewHoveredEventArgs {
+  preview: ISearchResultPreview;
+}
+
+export enum ResultPreviewsGridEvents {
+  PreviewHovered = 'previewHovered'
```

Comment:
Good idea, I'll make sure to add the `blur` event.

This was originally created since I had the wrong idea about the behaviour of `SuggestionsManager`. I thought the focus was meant to be in either `SuggestionsList` or `ResultPreviewsGrid` and I was going to use this method to detect mouse hover in `ResultPreviewsGrid` and blur `SuggestionsList`. This event shouldn't be necessary anymore, unless we want to detect mouse focus in `ResultPreviewsGrid` and prevent vertical arrow keys from changing the selected suggestion in `SuggestionsList`.